### ISSUE TITLE
Download a private file in GCS.

### ISSUE THREAD

**Issue by ANUBHAV2142** [googleapis/nodejs-storage#1291]
When i am trying to download a public file, there is no issue. But I want to download a private file as it what I need, but it is throwing "unauthorized"  401 error. It is printing 'Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object' message.
Please help me out or give me any alternative way to do it.
I have also tries with signed url but it only has read option to access the file and not download file.
So I am asking for your support, if you can provide a proper  solution for it that would be a great help.

**Comment by laljikanjareeya** [comment]
@ANUBHAV2142 can you provide the code snippet to better understand your problem.

You can refer the [sample](https://github.com/googleapis/nodejs-storage/blob/12f2133b7e3ec9612b3694d93b9ff756cb1bbd66/samples/downloadFile.js)  on how to download file using storage.

**Comment by ANUBHAV2142** [comment]
@laljikanjareeya I have already refered to the sample download file, and got the public download file using the same.
But it is not working for the private file.
I am getting this error "Error: Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object." as i have mentioned in earlier comment.
The code is same as the sample download file given.
It would be a great help if you can provide with any solution.
If you want i can post the complete code too, but it would be the exact same as in sample download file.

**Comment by laljikanjareeya** [comment]
@ANUBHAV2142 How you are making the file private?

If you did `file.makePrivate()` than only project maintainers can see and modify it.
while `file.makePrivate({ strict: true })` than only project owner can see and modify it

Here is complete demo on making file private and download private file.

```ts
async function yourFunction() {
  const storage = new Storage();
  const bucketName = 'your-bucket';
  const filename = `your-file.txt`;
  const bucket = storage.bucket(bucketName);
  await bucket.file(filename).makePrivate();

  // await bucket.file(filename).makePrivate({ strict: true }); This will only allow owner to see and modify.

  await bucket.file(filename).download({
    destination: 'path/your-file-name.txt'
  });
}
```

**Comment by ANUBHAV2142** [comment]
@laljikanjareeya I am not making my file private, the file I uploaded is by default not public(not publicly accessible) in the cloud storage bucket and I want to download it as it is. 
I tried your code and it is getting me "Error: Could not load the default credentials. Browse to https://cloud.google.com/docs/authentication/getting-started for more information.
    at GoogleAuth.getApplicationDefaultAsync".

**Comment by AVaksman** [comment]
@ANUBHAV2142 could you check what permissions you currently have.

It might be possible that you only have `storage.objects.create` but not `storage.objects.get` permission. As a `Storage Object Creator` role for example.

Please refer to the list of [permissions](https://cloud.google.com/storage/docs/access-control/iam-permissions#object_permissions) and list of [predefined roles](https://cloud.google.com/storage/docs/access-control/iam-roles)

**Comment by ANUBHAV2142** [comment]
@AVaksman Yes i am getting this error message "Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object".
But how to resolve this. I have editor permission on my service account. Should i change it to owner, will that resolve this error?

**Comment by AVaksman** [comment]
"Storage Object Admin" role should provide full control over objects:

> ![image](https://user-images.githubusercontent.com/7764119/93095909-b8c6e000-f671-11ea-8075-5aceb8781776.png)
